Increase minimum automake version to counteract faulty check_version function.
https://github.com/carlobaldassi/gimp-lqr-plugin/issues/11

Use newer automake if available.

Change nonportable echo -n to printf.
https://github.com/carlobaldassi/gimp-lqr-plugin/issues/12
https://github.com/carlobaldassi/gimp-lqr-plugin/pull/13

Disable maintainer mode.
--- autogen.sh.orig	2010-03-30 12:19:38.000000000 -0500
+++ autogen.sh	2024-08-04 02:49:08.000000000 -0500
@@ -13,7 +13,7 @@
 FILE=src/render.c
 
 AUTOCONF_REQUIRED_VERSION=2.54
-AUTOMAKE_REQUIRED_VERSION=1.6
+AUTOMAKE_REQUIRED_VERSION=1.10
 GLIB_REQUIRED_VERSION=2.0.0
 INTLTOOL_REQUIRED_VERSION=0.17
 
@@ -39,7 +39,7 @@
 
 DIE=0
 
-echo -n "checking for autoconf >= $AUTOCONF_REQUIRED_VERSION ... "
+printf "checking for autoconf >= %s ... " "$AUTOCONF_REQUIRED_VERSION"
 if (autoconf --version) < /dev/null > /dev/null 2>&1; then
     VER=`autoconf --version \
          | grep -iw autoconf | sed "s/.* \([0-9.]*\)[-a-z0-9]*$/\1/"`
@@ -52,8 +52,17 @@
     DIE=1;
 fi
 
-echo -n "checking for automake >= $AUTOMAKE_REQUIRED_VERSION ... "
-if (automake-1.7 --version) < /dev/null > /dev/null 2>&1; then
+printf "checking for automake >= %s ... " "$AUTOMAKE_REQUIRED_VERSION"
+if (automake-1.17 --version) < /dev/null > /dev/null 2>&1; then
+   AUTOMAKE=automake-1.17
+   ACLOCAL=aclocal-1.17
+elif (automake-1.16 --version) < /dev/null > /dev/null 2>&1; then
+   AUTOMAKE=automake-1.16
+   ACLOCAL=aclocal-1.16
+elif (automake-1.15 --version) < /dev/null > /dev/null 2>&1; then
+   AUTOMAKE=automake-1.15
+   ACLOCAL=aclocal-1.15
+elif (automake-1.7 --version) < /dev/null > /dev/null 2>&1; then
    AUTOMAKE=automake-1.7
    ACLOCAL=aclocal-1.7
 elif (automake-1.8 --version) < /dev/null > /dev/null 2>&1; then
@@ -85,7 +94,7 @@
     check_version $VER $AUTOMAKE_REQUIRED_VERSION
 fi
 
-echo -n "checking for glib-gettextize >= $GLIB_REQUIRED_VERSION ... "
+printf "checking for glib-gettextize >= %s ... " "$GLIB_REQUIRED_VERSION"
 if (glib-gettextize --version) < /dev/null > /dev/null 2>&1; then
     VER=`glib-gettextize --version \
          | grep glib-gettextize | sed "s/.* \([0-9.]*\)/\1/"`
@@ -98,7 +107,7 @@
     DIE=1
 fi
 
-echo -n "checking for intltool >= $INTLTOOL_REQUIRED_VERSION ... "
+printf "checking for intltool >= %s ... " "$INTLTOOL_REQUIRED_VERSION"
 if (intltoolize --version) < /dev/null > /dev/null 2>&1; then
     VER=`intltoolize --version \
          | grep intltoolize | sed "s/.* \([0-9.]*\)/\1/"`
@@ -130,7 +139,7 @@
 echo
 echo "I am going to run ./configure with the following arguments:"
 echo
-echo "  --enable-maintainer-mode $AUTOGEN_CONFIGURE_ARGS $@"
+echo "  $AUTOGEN_CONFIGURE_ARGS $@"
 echo
 
 if test -z "$*"; then
